service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java

```diff
@@ -71,6 +73,33 @@ private static String getRoleSessionName(String value) {
         : value;
   }
 
+  public static void appendUserTags(Collection<Tag> tags, AuthenticatedUserRequest userRequest) {
+    tags.add(Tag.builder().key("user_id").value(userRequest.getSubjectId()).build());
+  }
+
+  public static void appendResourceTags(Collection<Tag> tags, AwsCloudContext awsCloudContext) {
+    tags.add(Tag.builder().key("version").value(awsCloudContext.getMajorVersion()).build());
+    tags.add(Tag.builder().key("tenant").value(awsCloudContext.getTenantAlias()).build());
+    tags.add(Tag.builder().key("environment").value(awsCloudContext.getEnvironmentAlias()).build());
+  }
+
+  public static void appendPrincipalTags(
+      Collection<Tag> tags,
+      AwsCloudContext awsCloudContext,
+      ControlledAwsStorageFolderResource awsStorageFolderResource) {
+    tags.add(Tag.builder().key("version").value(awsCloudContext.getMajorVersion()).build());
```

Comment:
The newer environment deployments (including `devel`) use the more conventional camel casing for the tag keys (defined as the values [here](https://github.com/verily-src/mc-terra-aws-wsm-landingzone/blob/main/modules/environment/common.tf#L10)):

* `version -> Version`
* `bucket_id -> S3BucketID`
* `terra_bucket_id -> TerraBucketID`